### PR TITLE
Add bool type

### PR DESCRIPTION
**Pull request by waterlink**
This PR:
- first extracts `Model::Fields` to the file of its own
- adds missing `#inspect` for `Null` types
- finally adds support for `Bool` type

### CODE REVIEW

**Comment by waterlink** [comment]
@sdogruyol Can you take a look at the code of this PR?

File: src/types.cr [hunk 1]
```diff
@@ -32,6 +32,10 @@ struct Int
       io << ""
     end
 
+    def inspect
```
Comment (sdogruyol) [inline src/types.cr]: Isn't there a generic way of inspecting this Null objects ( something like a macro in top level )

File: src/model.cr [hunk 2]
```diff
@@ -1,84 +1,15 @@
 require "./criteria_helper"
 require "./support"
 
+class ActiveRecord::Model; end
+
+require "./model/fields"
```
Comment (sdogruyol) [inline src/model.cr:6]: I like the seperation of Fields :+1:

File: src/types.cr (hunk 1 above)
Comment (waterlink) [inline src/types.cr]: Oh, you are right, this can be DRYed up a bit, thanks!

Comment (waterlink) [inline src/types.cr]: Fixed.

**Comment by waterlink** [comment]
@sdogruyol Can you take another look?

File: src/types.cr [hunk 3]
```diff
@@ -13,6 +13,10 @@ macro active_record_define_not_null_for(type, name)
         ::raise ActiveRecord::NullCheckFailed.new("It is {{name.id}}::Null")
       end
 
+      def inspect
+        "Null(#{{{type.id.stringify}}})"
```
Comment (sdogruyol) [inline src/types.cr:17]: :+1:

**Comment by sdogruyol** [comment]
@waterlink Looks great :+1:

**Comment by waterlink** [comment]
@sdogruyol Can you take another look, with last commit I have added, now `fields.cr` looks much better to me.

**Comment by waterlink** [comment]
Documentation sounds good to me. I will write it a bit later (maybe today, or tomorrow).

**Comment by waterlink** [comment]
@sdogruyol I have added documentation to `Model::Fields`. Can you take another look?

**Comment by sdogruyol** [comment]
LGTM :+1:

**Pull request merged**